lambdas/third-party-healthcheck/src/main/java/uk/gov/di/ipv/cri/kbv/api/handler/ThirdPartyHealthCheckHandler.java

```diff
@@ -0,0 +1,206 @@
+import software.amazon.lambda.powertools.logging.CorrelationIdPathConstants;
+import software.amazon.lambda.powertools.logging.Logging;
+import software.amazon.lambda.powertools.metrics.Metrics;
+import software.amazon.lambda.powertools.tracing.Tracing;
+import uk.gov.di.ipv.cri.common.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.cri.common.library.service.ConfigurationService;
+import uk.gov.di.ipv.cri.kbv.api.service.ServiceFactory;
+import uk.gov.di.ipv.cri.kbv.api.tests.ErrorReport;
+import uk.gov.di.ipv.cri.kbv.api.tests.Test;
+import uk.gov.di.ipv.cri.kbv.api.tests.assertions.AssertionTest;
+import uk.gov.di.ipv.cri.kbv.api.tests.keystore.KeyStoreTest;
+import uk.gov.di.ipv.cri.kbv.api.tests.keytool.ImportCertificateTest;
+import uk.gov.di.ipv.cri.kbv.api.tests.soap.SOAPRequestTest;
+import uk.gov.di.ipv.cri.kbv.api.tests.soap.report.LoginWithCertificateTestReport;
+import uk.gov.di.ipv.cri.kbv.api.tests.ssl.SSLHandshakeTest;
+import uk.gov.di.ipv.cri.kbv.api.tests.ssl.report.SSLHandshakeTestReport;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ThirdPartyHealthCheckHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOGGER = LogManager.getLogger(ThirdPartyHealthCheckHandler.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final int SUCCESS_HTTP_CODE = 200;
+    private static final int FAIL_HTTP_CODE = 525;
+
+    private final List<Test<?>> testSuits;
+
+    @ExcludeFromGeneratedCoverageReport
+    public ThirdPartyHealthCheckHandler() throws IOException {
+        this(new ServiceFactory());
+    }
+
+    public ThirdPartyHealthCheckHandler(ServiceFactory serviceFactory) throws IOException {
+        ConfigurationService configurationService =
+                new ConfigurationService(
+                        serviceFactory.getSsmProvider(), serviceFactory.getSecretsProvider());
+
+        String waspUrl = configurationService.getSecretValue(Configuration.WASP_URL_SECRET);
+        String keystorePassword =
+                configurationService.getSecretValue(Configuration.KEYSTORE_PASSWORD);
+        String keystoreSecret = configurationService.getSecretValue(Configuration.KEYSTORE_SECRET);
+
+        LOGGER.info("WASP URL: {}", waspUrl);
+        LOGGER.info("keystorePassword: {}", keystorePassword);
+        LOGGER.info("keystoreSecret: {}", keystoreSecret);
+
+        this.testSuits =
+                List.of(
+                        new ImportCertificateTest(keystoreSecret, keystorePassword),
+                        new SSLHandshakeTest(Configuration.WASP_HOST, Configuration.WASP_PORT),
+                        new SOAPRequestTest(keystoreSecret, keystorePassword, waspUrl),
+                        new KeyStoreTest(keystoreSecret, keystorePassword));
```

Comment:
It probably better to inject this list into the constructor, it would make this thing easier to test.

Reply:
Good shout - I was meant to change this to varargs but forgot to :p